src/AzureAuth.Test/Ado/AdoTokenTest.cs

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.AzureAuth.Test
+{
+    using FluentAssertions;
+
+    using Microsoft.Authentication.AzureAuth.Ado;
+    using Microsoft.Office.Lasso.Interfaces;
+
+    using Moq;
+
+    using NUnit.Framework;
+
+    public class AdoTokenTest
+    {
+        private const string SystemAT = "SYSTEM_ACCESSTOKEN";
+        private const string AzureAuthADOPAT = "AZUREAUTH_ADO_PAT";
+        private Mock<IEnv> mockEnv;
+
+        [SetUp]
+        public void SetUp()
+        {
+            this.mockEnv = new Mock<IEnv>(MockBehavior.Strict);
+        }
+
+        [Test]
+        public void No_PAT()
+        {
+            this.mockEnv.Setup(e => e.Get(AzureAuthADOPAT)).Returns<string>(default);
+            this.mockEnv.Setup(e => e.Get(SystemAT)).Returns<string>(default);
+
+            AdoToken.PatResult expected = new()
+            {
+                Exists = false,
+                EnvVarSource = null,
+                Value = null,
+            };
+
+            AdoToken.PatFromEnv(this.mockEnv.Object).Should().Be(expected);
+        }
+
+        [Test]
+        public void EmptyPAT()
+        {
+            this.mockEnv.Setup(e => e.Get(AzureAuthADOPAT)).Returns(string.Empty);
+            this.mockEnv.Setup(e => e.Get(SystemAT)).Returns(string.Empty);
+
+            AdoToken.PatResult expected = new()
+            {
+                Exists = false,
+                EnvVarSource = null,
+                Value = null,
+            };
+
+            AdoToken.PatFromEnv(this.mockEnv.Object).Should().Be(expected);
+        }
+
+        [Test]
+        public void Returns_First_PAT_From_EnvVar_List()

```

Comment:
I think it might be better to frame these tests according to the name of the environment variable rather than "first PAT" and "second PAT". The EnvVar List is private and merely an implementation detail.

It might also be better to frame these tests as `AZUREAUTH_ADO_PAT` taking precedence over `SYSTEM_ACCESSTOKEN`.

I would name this test `Returns_AZUREAUTH_ADO_TOKEN` and the next test as `Returns_SYSTEM_ACCESSTOKEN_When_AZUREAUTH_ADO_TOKEN_Not_Defined` (or similar). 🤷‍♂️